Blackjack/Game/BlackjackGame.m

```diff
@@ -0,0 +1,254 @@
+//
+//  BlackjackGame.m
+//  Blackjack
+//
+//  Created by Alex on 28/09/2022.
+//
+
+#import "BlackjackGame.h"
+#import "PlayingCard.h"
+
+typedef NS_ENUM(NSInteger, State) {
+    COLLECT_BETS, DEAL_CARDS, AWAITING_PLAYERS_DECISION, AWAITING_DEALER
+};
+
+// MARK: - Private Properties
+
+@interface BlackjackGame ()
+
+@property (nonatomic, strong) Deck *deck;
+@property (nonatomic) NSInteger cardsAmountToWin;
+@property (nonatomic) NSInteger dealerMinimumCardEvaluation;
+@property (nonatomic, assign) State gameState;
+@property (nonatomic) NSArray *gameStates;
+
+@end
+
+// MARK: - Implementation
+
+@implementation BlackjackGame
+
+@synthesize dealer = _dealer; // because we provide setter AND getter
+
+- (Contestant *) dealer {
+    if (!_dealer)
+        _dealer = [[Contestant alloc] initWith:@"Dealer" cards:[NSMutableArray new] chips: 0 isPlaying:YES];
+    return _dealer;
+}
+
+// MARK: - Static Methods
+
++ (NSDictionary<NSString*, NSNumber*> *)cardValues {
+    return @{@"2": @2, @"3": @3, @"4": @4, @"5": @5, @"6": @6, @"7": @7, @"8": @8, @"9": @9, @"10": @10, @"J": @10, @"Q": @10, @"K": @10};
+}
+
++ (NSInteger)cardsAmountToWin {
+    return 21;
+}
+
++ (NSInteger)dealerMinimumCardEvaluation {
+    return 16;
+}
+
+- (NSArray *)currentActivePlayers {
+    NSMutableArray *activePlayers = [NSMutableArray new];
+    for (Player *player in self.players) {
+        if (player.isPlaying) {
+            [activePlayers addObject:player];
+        }
+    }
+    return activePlayers;
+}
+
+// MARK: - Init
+
+- (instancetype)initWithDeck:(Deck *)deck {
```

Comment:
You have too many properties uninitiated in your init with no reason...

`dealer` should be instantiated here and not in a lazy fashion, as there is no game without a dealer.

same goes for your `players` array and set the `currentActivePlayer` from that.

In addition, when possible, we prefer to receive the `delegate` in our `init` so that it will be initialized.

In other words...
Any property you can initialize in your `init`, do so.

Reply:
From past experience "delegate" is not a dependency therefor should not be passed in the init.
Having access to the delegate property allows to change (e.x tableView.delegate)
What do you think?

Reply:
I think that you have no intention of changing your delegate, and in such a case you should not allow this.

I can only see very little use cases in which this is true.

I know Apple does this in a lot of places, but that doesn't make it good in my opinion.

Reply:
Done

Reply:
Seems like your delegate is still not read-only, although you only set it in your constructor...